services/harmony/app/middleware/cmr-collection-reader.ts

```diff
@@ -122,7 +122,9 @@ async function cmrCollectionReader(req: HarmonyRequest, res, next: NextFunction)
       // The request used a short name
       const shortNameMatch = req.url.match(COLLECTION_ROUTE_REGEX);
       if (shortNameMatch) {
-        const shortName = shortNameMatch[1].substr(1, shortNameMatch[1].length - 2);
+        const shortNamePart = shortNameMatch[1].substr(1, shortNameMatch[1].length - 2);
+        // fix the short name for ogc EDR requests
+        const shortName = shortNamePart.replace(/^ogc-api-edr\/1\.0\.0\/collections\//, '');
```

Comment:
Yeah this is one of those things I was wondering about - we're making several little tweaks to try to use the same regex for different routes. It might be best to have separate functions for extracting the collection/short name for each type of route instead rather than trying to force the COLLECTION_ROUTE_REGEX to try to handle everything.